javalib/src/main/scala/java/net/InetAddress.scala

```diff
@@ -38,8 +38,10 @@ import scala.scalanative.meta.LinktimeInfo.{isLinux, isMac}
  *    That information is handled separately.
  */
 
-class InetAddress protected (ipAddress: Array[Byte], originalHost: String)
-    extends Serializable {
+class InetAddress protected (
```

Comment:
I can do another line-by-line pass but after quite a bit
of study, I do not understand why this file is being touched
at all.  There is always the chance that I made errors but
I recommend this PR focus on UDP issues and any errors I made be done in separate, easier to review PRs.

Reply:
Why did the `val ipAddress` go away in `Inet6Address` and then get introduced to `InetAddress`? Seems they should at least be consistent.

Reply:
I just moved the member definition to the upper class. In `Inet6Address` and `Inet4Address`, we just have constructor parameter which should be enough. Implementation consistency on the sub-classes is better than it used to be.

Reply:
I think these two `Inet?Address` changes are exactly the kind of thing which should be discussed in
another Issue/PR set.  I can create such.  

These two changes have nothing to do with `Datagram`.  Extensive discussion here will only
distract and delay this PR.

Based on the discussion here, I did a small private design study to address the concern 
and can post it as a separate PR. 
That will keep the focus here on on `Datagram` and there on `Inet?Address`.

Sound like a plan?

Reply:
Agreed. We can totally extract those inet address related changes in another PR. Let me know once you are done with it, I'll rebase the PR afterward.

Reply:
re: We can totally extract those inet address related changes in another PR

PRs #3654 & #3655 are in CI now.  So far, so good. Looks like CI will take hours to finish.
I will check completion by daylight.

Do those PRs capture the changes we have been discussing? 

re: Let me know once you are done with it, I'll rebase the PR afterward.

Unless I have missed a change in this bigger-than-a-breadbox PR, I think
you do not have to wait for merge of those two PRs and rebase. I think
that you can delete the two Inet?Address files from this PR.  That should
cause a re-run of CI.  If successful, you can then promote to full PR.